boardgame-ui/src/components/timer/AddTimerForm.vue

```diff
@@ -0,0 +1,298 @@
+    },
+
+    timerTypeI18nPath() {
+      return [
+        {type: TimerTypes.COUNT_UP, i18nPath: 'timer.type.count_up'},
+        {type: TimerTypes.COUNT_DOWN, i18nPath: 'timer.type.count_down'},
+        {type: TimerTypes.RELOAD, i18nPath: 'timer.type.reload'}
+      ];
+    }
+  },
+
+  watch: {
+    initial_duration_seconds: function(val) {
+      this.timer.initial_duration = val * 1000;
+    },
+    reload_increment_seconds: function(val) {
+      this.timer.reload_increment = val * 1000;
+    },
+  },
+
+  methods: {
+    selectBoardGame(option) {
+      this.timer.id_board_game = option ? option.id : null;
+    },
+
+    addPlayer() {
+      this.players.push({user: null, color: this.generateRandomColor()});
+    },
+
+    generateRandomColor() {
+      return '#' + (~~(Math.random() * (1 << 24))).toString(16);
+    },
+
+    removePlayer(idx) {
+      if(this.players.length === 1) {
+        this.$toast.open({
+          message: this.$t('add-edit-game.must-have-at-least-one-player'),
+          type: 'is-danger',
+          position: 'is-bottom'
+        });
+        return;
+      }
+      this.players.splice(idx, 1);
+    },
+
+    async createTimer(scope) {
+      let result = await this.validate(scope);
+
+      if (!result) {
+        return;
+      }
+
+      this.timer.player_timers = [];
+      for (let key in this.players) {
+        let player = this.players[key];
+        if (player.user.id != null) {
+          this.timer.player_timers.push({id_user: player.user.id, name: null, color: player.color});
+        }
+        else {
+          this.timer.player_timers.push({id_user: null, name: player.user.name, color: player.color});
+        }
+      }
+
+      try {
+        await this.timer.save();
+        this.$emit('timerCreated', this.timer);
+      }
+      catch (e) {
+        console.log(e);
+      }
+    },
+
+    async validate(scope) {
+      let result = await this.$validator.validateAll(scope);
+
+      if (!result) {
+        this.$toast.open({
+          message: this.$t('global.invalid-form'),
+          type: 'is-danger',
+          position: 'is-bottom'
+        });
+      }
+
+      return result;
+    }
+  },
+
+  async created() {
+    this.isLoading = true;
+    this.allUsers = await User.fetchUsers();
+
+    this.boardGames = await BoardGame.fetchAll();
+
+    if (this.$route.params.id) {
+      this.timer = await Timer.fetch(this.$route.params.id);
+
+      // initialize for models
+      this.initial_duration_seconds = Math.floor(this.timer.initial_duration / 1000);
+      this.reload_increment_seconds = this.timer.reload_increment ? Math.floor(this.timer.initial_duration / 1000) : 0;
+
+      for (let key in this.timer.player_timers) {
+        let player = this.timer.player_timers[key];
+        this.players.push({user: player.user, color: player.color});
+      }
+
+      if (this.timer.id_board_game != null) {
+        this.searchString = this.boardGames.find(function(element) {
```

Comment:
```suggestion
        this.searchString = this.boardGames.find(element => {
```
If not using arrow function, I get an "this is undefined" error

Reply:
👍